types/persist/persist.go

```diff
@@ -89,6 +90,7 @@ func (p *Persist) Equals(p2 *Persist) bool {
 		p.UserProfile.Equal(&p2.UserProfile) &&
 		p.NetworkLockKey.Equal(p2.NetworkLockKey) &&
 		p.NodeID == p2.NodeID &&
+		p.AttestationKey == p2.AttestationKey &&
```

Comment:
this will only match if both are nil or if both are the exact same pointer.
but it will not match if it's the same underlying key loaded separately.

you could compare the public keys here (assuming both attestation keys are non-nil), since they should always implement the `Equal` method.